event.go

```diff
@@ -0,0 +1,110 @@
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+package rapid
+
+import (
+	"log"
+	"sort"
+)
+
+// counter maps a (stringified) event to a frequency counter
+type counter map[string]int
+
+// stats maps labels to counters
+type stats map[string]counter
+
+type event struct {
+	label string
+	value string
+}
+type done struct {
+	result chan stats
+}
+
+type counterPair struct {
+	frequency int
+	event     string
+}
+
+// Event records an event for test `t` and
+// stores the event for calculating statistics.
+//
+// Recording events and printing a their statistic is a tool for
+// analysing test data generations. It helps to understand if
+// your customer generators produce value in the expected range.
+//
+// Each event has a label and an event value. To see the statistics,
+// run the tests with `go test -v`.
+//
+func Event(t *T, label string, value string) {
+	t.Helper()
+	if t.evChan == nil {
+		log.Printf("Creating the channels for test %s", t.Name())
+		t.evChan = make(chan event)
+		t.evDone = make(chan done)
```

Comment:
This is not thread-safe. Also, I don't think there is a good reason to use channels/goroutings: `sync.Mutex` will most likely result in simpler (and faster) code.

Reply:
Argh, you are right. That is indeed not thread safe.

Reply:
Now using a mutex. Other open stuff will come in the next days.